src/plugins/upf/upf_classify.c

```diff
@@ -170,7 +200,14 @@ upf_classify (vlib_main_t * vm, vlib_node_runtime_t * node,
 		    {
 		      vnet_buffer (b)->gtpu.pdr_idx = pdr - active->pdr;
 		      far = sx_get_far_by_id(active, pdr->far_id);
+
```

Comment:
Most likely is this not the right place to integrate the DPI scanning into the classify path....

Reply:
Even after some changes, this still feels wrong.

If we have already detected a flow, there should a point in the life of the flow where we know which PDR, FAR and URR applies to it. Redoing all the PDR and FAR matching and lookup should be skipped at that point in time. I'm missing the logic that handles this.